packages/core/src/Parser.ts

```diff
@@ -390,10 +446,33 @@ export const Parser = {
             } while (context);
         }
 
+        const selection = node.ownerDocument.getSelection();
+        const range = selection.rangeCount && selection.getRangeAt(0);
+        if (range) {
+            const isForward = range.startContainer === selection.anchorNode;
+            const vRange = parseRange({
+                startContainer: range.startContainer,
+                startOffset: range.startOffset,
+                endContainer: range.endContainer,
+                endOffset: range.endOffset,
+                direction: isForward ? Direction.FORWARD : Direction.BACKWARD,
+            });
+            if (vRange) {
+                vDocument.range.set(vRange);
+            }
+        }
+
         if (!vDocument.range.start.parent || !vDocument.range.end.parent) {
             vDocument.range.setAt(vDocument.root);
         }
 
         return vDocument;
     },
+    /**
+     * Convert the DOM values for a range to set to VRange locations in the
+     * CustomEvent's payload.
+     *
+     * @param range
+     */
+    parseRange: parseRange,
```

Comment:
It's funny because the comments are different between here and the function itself... ^^'